src/store/RelayStoreData.js

```diff
@@ -86,10 +84,7 @@ class RelayStoreData {
    * Get the data set backing actual Relay operations. Used in GraphQLStore.
    */
   static getDefaultInstance(): RelayStoreData {
-    if (!_instance) {
-      _instance = new RelayStoreData();
-    }
-    return _instance;
+    return RelayStore.getStoreData();
```

Comment:
The motivation was to get rid of `storeData` argument in the constructor of `RelayContext`. That makes us sure that no two instances of `RelayContext` share the same instance of `RelayStoreData`, so it is enough to write `relayContextA !== relayContextB` to check that `relayContextA.getStoreData() !== relayContextB.getStoreData()`.

I think, we could break the circular dependency by removing `getDefaultInstance` from `RelayStoreData`, and replacing `RelayStoreData.getDefaultInstance()` with `RelayStore.getStoreData()` everywhere (it's only 3 modules and 2 tests in OSS). What do you think?